internal/services/v1/watch.go

```diff
@@ -26,16 +28,31 @@ type watchServer struct {
 	v1.UnimplementedWatchServiceServer
 	shared.WithStreamServiceSpecificInterceptor
 
-	heartbeatDuration time.Duration
+	config WatchServerConfig
+}
+
+type WatchServerConfig struct {
+	HeartbeatDuration time.Duration
+
+	// StreamingAPITimeout is the timeout for streaming APIs when no response has been
+	// recently received.
+	StreamingAPITimeout time.Duration
 }
 
 // NewWatchServer creates an instance of the watch server.
-func NewWatchServer(heartbeatDuration time.Duration) v1.WatchServiceServer {
+func NewWatchServer(config WatchServerConfig) v1.WatchServiceServer {
+	configWithDefaults := WatchServerConfig{
+		HeartbeatDuration:   config.HeartbeatDuration,
+		StreamingAPITimeout: defaultIfZero(config.StreamingAPITimeout, 30*time.Second),
```

Comment:
let's see a much higher default for watch timeout; perhaps an hour?

Reply:
i'd love to, but the problem with that is that the users may be misled by our docs.

according to ` go run ./cmd/spicedb/main.go serve --help`:

```
 --streaming-api-response-delay-timeout duration           
maximum time until the server will return an error to a caller of a streaming API because no data has been sent (default 30s)
```

so unless we split this flag into two, or use the same default everywhere, i'd keep this as is.

Reply:
I'd say we split the flag into two and have one for `watch`

Reply:
I'm not into that idea because the `serve` command already haves a lot of flags https://authzed.com/docs/spicedb/concepts/commands#reference-spicedb-serve

we should have a very good reason to add another one.

why do we need this flag in the first place? could we not replace it with guidance that if deploying spicedb in prod, it should be behind a load balancer that puts timeouts in place?

Reply:
Because watch is intended to be a very long running process and other streams are not; a LookupResources that takes over 45 seconds to return is a problem. A watch that closes before minutes or even hours is up, is unexpected.

Reply:
Okay, on second thought, we don't need this PR at all 😆 i can just amend the description of the flag to mention the ones it affects